### DISCUSSION TITLE
Add more ANSI keys to 'Key' enum?

### DISCUSSION THREAD

**Discussion by AlbatorLaho** [tui-cs/Terminal.Gui#1998] [Q&A]
Is there plans to add more keys to the 'Key' enum such as comma, period, brackets, question mark, plus, minus, &c.?

I am working on a project where the key bindings are serialized to a file.
But since the enum doesn't define names for the above mentioned keys (and others), it just uses the int value instead.
It's more or less fine, but my hope was that the user could easily edit the file without having to know what int value to use.

**Comment by tznind** [comment]
This is a great suggestion

For reference here is a list that appears in the `System.Window.Inut` namespace:
https://docs.microsoft.com/en-us/dotnet/api/system.windows.input.key?view=windowsdesktop-6.0 

And the winforms (`System.Windows.Forms`) one:
https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.keys?view=windowsdesktop-6.0

@BDisp can you see any issue with adding these to [Keys enum](https://github.com/gui-cs/Terminal.Gui/blob/develop/Terminal.Gui/Core/Event.cs#L61)?

If yes then we should go through the list I linked and identify those that are consistently sent through the Terminal when pressed (e.g. in PowerShell, Bash etc).  I did look to see if I could find a terminal spec (e.g. VT100) which lists all the supported normal keys but couldn't find one easily (I just kept running into docs on escape codes etc) and theres always peculiarities like ambigious codes.  But for most of these punctuation keys it shouldn't be a problem.

**Comment by BDisp** [comment]
The `System.Window.Input` has values that not correspond to the https://www.asciitable.com/. See the numeric values from `D0 to D9` (from 34 to 43) and the ASCII table is from 48 to 57.
The `System.Windows.Forms` is more reliable.
The integration to the `Terminal.Gui` respects the ASCII char codes. The particularity is not colliding with some keys like:
`CharMask, SpecialMask, ShiftMask, AltMask, CtrlMask`
Valid inputs are checked by these keys and are very tied to the ProcessKey. In my opinion some of theses keys should not belong to the `Key` enum, but controlled by the `KeyModifiers` class. The problem is the check validation are very tied on to the `Key` enum that not very easy to change this behavior.
But the @Albator11  suggestion is doable for some commons keys.

**Comment by tig** [comment]
Someone please create an Issue to track this?

**Reply by tznind** [reply]
Done: https://github.com/gui-cs/Terminal.Gui/issues/2088
